### DISCUSSION TITLE
我想给apollo-java 客户端，加一个拉取不同appid的配置。你们官方会接纳么

### DISCUSSION THREAD

**Discussion by TerryLam2010** [apolloconfig/apollo-java#65] [Ideas]
我想给apollo-java 客户端，加一个拉取不同appid的配置，并支持secret配置。你们官方会接纳么

**Comment by nobodyiam** [comment]
这个 feature 社区中有一些场景是需要的，不过现有的 apollo-client 在设计上是围绕一个 appid 实现的，能描述下你的实现思路吗？是新写一个 client 还是重构现有的 client？

**Reply by TerryLam2010** [reply]
我是打算在你们现有apollo-client中重构提供这个功能。
bootstrap: 仍然只根据主appid+namespace拉取数据
配置：appid+namespace 进行分组，获取到的配置进行分组后，注解上也会有相应的appid和namespace配置，用于进行反射
1.改造AbstractConfigRepository下的实现，根据多个appid+namespace 调用接口获取配置。当然RemoteConfigLongPollService作为获取变更通知也得按appid+namespace 分组，appid+namespace 为一个线程做监听。
2.ConfigService 也得支持appid+namespace 获取配置
3.@ApolloConfig，@ApolloJsonValue 需要支持appid+namespace
4.apollo.accesskey.secret.multiple=xxxx/yyyy,xxxx/yyyy  xxxx代表为一个appid,yyy代表秘钥
5.补充，当然还有其他很多类需要加上appid+namespace 做适配。

**Reply by nobodyiam** [reply]
大方向我觉得可以的，不过对于存量的 api/配置等需要考虑透明兼容，建议可以先写一个设计文档社区一起 review 下~

**Reply by TerryLam2010** [reply]
直接发在这里么？还是你们有个群？

**Reply by nobodyiam** [reply]
可以在这个项目中开一个 issue，这样大家都可以 review

**Reply by TerryLam2010** [reply]
如果可以的话，能尽快敲定最好。我需要在某个版本进行改造打包（自己用），并在最新版本提交pr。
